zebra-test/src/command.rs

```diff
@@ -15,9 +15,19 @@ pub fn test_cmd(path: &str) -> Result<(Command, impl Drop)> {
     let mut cmd = Command::new(path);
     cmd.current_dir(dir.path());
 
-    let state_dir = dir.path().join("state");
-    fs::create_dir(&state_dir)?;
-    env::set_var("ZEBRAD_CACHE_DIR", state_dir);
+    let cache_dir = dir.path().join("state");
+    fs::create_dir(&cache_dir)?;
+
+    fs::File::create(dir.path().join("zebrad.toml"))?.write_all(
+        format!(
+            "[state]\ncache_dir = '{}'",
+            cache_dir
+                .into_os_string()
+                .into_string()
+                .map_err(|_| eyre!("tmp dir path cannot be encoded as UTF8"))?
+        )
+        .as_bytes(),
+    )?;
```

Comment:
I don't feel like we should be using serialized toml files to construct configs, I'd much prefer to have a function for creating a `Config` file and one for writing that `Config` to the disk as toml.

Reply:
This seemed much simpler, and avoids creating a cyclic dependency between `zebra-test` and `zebrad`.  In the future this could be resolved by moving this `zebrad`-specific test code into the `zebrad` crate, where the `Config` is available.  But I would prefer not to do restructuring of the test code in this PR.

Reply:
I'm not too concerned about using fixed configs in tests.

Worse case scenario, the tests fail because we change the config format, and we fix the tests.